pulser-core/pulser/devices/_device_datacls.py

```diff
@@ -464,8 +471,8 @@ def _to_dict(self) -> dict[str, Any]:
     def _to_abstract_repr(self) -> dict[str, Any]:
         defaults = get_dataclass_defaults(fields(self))
         params = self._params()
-        for p in ALWAYS_OPTIONAL_PARAMS:
-            if params[p] == defaults[p]:
+        for p in OPTIONAL_IN_ABSTR_REPR:
+            if p in params and params[p] == defaults[p]:
```

Comment:
What value is covered by "p in params" ?

Reply:
`accepts_new_layouts`